src/modules/videoconference/VideoConferenceModule/Toolbar.cpp

```diff
@@ -154,18 +154,19 @@ LRESULT Toolbar::WindowProcessMessages(HWND hwnd, UINT msg, WPARAM wparam, LPARA
         static bool previousShow = false;
         bool show = false;
 
-        if (toolbar->cameraInUse)
+        if (toolbar->ToolbarHide == L"Never")
         {
-            show = toolbar->HideToolbarWhenUnmuted ? toolbar->microphoneMuted || toolbar->cameraMuted : true;
+            show = true;
         }
-        else if (toolbar->previouscameraInUse)
+        else if (toolbar->ToolbarHide == L"When both camera and microphone are muted")
         {
-            VideoConferenceModule::unmuteAll();
```

Comment:
We need to keep this logic to unmute devices once the camera is no longer in use

Reply:
Will this not catch the user off guard? Should we really be muting or unmuting things automatically?

Reply:
Sorry for the late response. I was sick these past few weeks.

I had an idea when I started taking a look at it again. Tell me if this sounds intuitive to you
1. Hides when both camera and microphone are muted
  * If the camera was previously not in use then everything automatically gets muted(or maybe it should just change the camera's state)
  * If the camera was previously in use then nothing changes
2. Hides when both camera and microphone are unmuted
  * If the camera was previously not in use everything gets unmuted(or maybe it should just change the camera's state)
  * If the camera  was previously in use nothing changes

I have not implemented any of this, but if you agree I can move forward with the implementation